### PR TITLE
KAZOO-4089: fix quickcall for users

### PR DESCRIPTION
**Pull request by fenollp**

### CODE REVIEW

File: applications/crossbar/src/modules/cb_modules_util.erl [hunk 1]
```diff
@@ -288,7 +289,7 @@ originate_quickcall(Endpoints, Call, Context) ->
     Request = [{<<"Application-Name">>, <<"transfer">>}
                ,{<<"Application-Data">>, get_application_data(Context)}
                ,{<<"Msg-ID">>, MsgId}
-               ,{<<"Endpoints">>, maybe_auto_answer(Endpoints, AutoAnswer)}
+               ,{<<"Endpoints">>, set_outbound_call_id(CallId, maybe_auto_answer(Endpoints, AutoAnswer))}
```
Comment (jamesaimonetti) [inline applications/crossbar/src/modules/cb_modules_util.erl]: Maybe you could refactor this to not iterate over the list of endpoints twice? Convert it to a fold and run the `set_value` calls as functions to the fold?

File: applications/crossbar/src/modules/cb_modules_util.erl [hunk 2]
```diff
@@ -327,11 +331,13 @@ get_application_data_from_nouns(_Nouns) ->
 
 -spec get_timeout(cb_context:context()) -> pos_integer().
 get_timeout(Context) ->
-    try wh_util:to_integer(cb_context:req_value(Context, <<"timeout">>, 30)) of
-        Timeout when Timeout > 3 -> Timeout;
-        _ -> 30
+    Default = 30,
```
Comment (jamesaimonetti) [inline applications/crossbar/src/modules/cb_modules_util.erl]: A macro may be more descriptive `?DEFAULT_TIMEOUT_S` to indicate the unit of time is seconds here.

File: core/kazoo_documents-1.0.0/src/kz_account.erl [hunk 3]
```diff
@@ -156,12 +156,13 @@ is_superduper_admin(JObj) ->
 allow_number_additions(JObj) ->
     wh_json:is_true(?ALLOW_NUMBER_ADDITIONS, JObj).
 
--spec fetch(ne_binary()) -> {'ok', doc()} |
+-spec fetch(api_binary()) -> {'ok', doc()} |
                             {'error', _}.
+fetch('undefined') ->
```
Comment (jamesaimonetti) [inline core/kazoo_documents-1.0.0/src/kz_account.erl]: Seems like defensive code, non?

**Comment by jamesaimonetti** [comment]
Hope you didn't manually sort those lists of headers :) Thanks for doing it

File: core/kazoo_documents-1.0.0/src/kz_account.erl (hunk 3 above)
Comment (fenollp) [inline core/kazoo_documents-1.0.0/src/kz_account.erl]: It's actually to reflect behaviour that used to exist before. I mean the switch from open_cache_doc to fetch removed this "dont crash on undefined, but return error" behaviour.

**Comment by fenollp** [comment]
@jamesaimonetti `M-x sort-lines`

File: core/kazoo_documents-1.0.0/src/kz_account.erl (hunk 3 above)
Comment (fenollp) [inline core/kazoo_documents-1.0.0/src/kz_account.erl]: @k-anderson You said to backport this part, right?

Comment (fenollp) [inline core/kazoo_documents-1.0.0/src/kz_account.erl]: @k-anderson ping

**Pull request merged**